intro/integration/js.md

```diff
@@ -23,44 +29,141 @@ import skygear from 'skygear/react-native'
 ```
 
 
-# Using older version
+# Configurating
+
+You can use the following code snippet to configure the SDK to connect to server.
+
+```
+skygear.config({
+  'endPoint': 'https://<your-app-name>.skygeario.com/', // trailing slash is required
+  'apiKey': '<your-api-key>',
+}).then(() => {
+  console.log('skygear container is now ready for making API calls.');
+}, (error) => {
+  console.error(error);
+});
+```
+
+## Angular and Ionic
+
+A project using Angular (including Ionic) should follow Angular's pattern of managing data. Skygear is usually used as a global singleton object, the Angular way to manage such data is through [Service](https://angular.io/docs/ts/latest/tutorial/toh-pt4.html).
 
-For versions before 0.24, some manual works are needed for integrating frameworks building with webpack, popular examples are Ionic and Angular.
+First, you need to create a `SkygearService` that hold an refernce of the singleton skygear object. An empty service looks like this, save this as `skygear.service.ts`.
+
+```
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class SkygearService {
+}
+```
+
+Then you have to add a method to the service for configurating and returing the skygear instance.
+
+```
+  isConfigurated = false;
+  getSkygear() {
+    if (this.isConfigurated) {
+      // Return the instance directly if it is ready
+      return Promise.resolve(skygear);
+    }
+    let promise = skygear.config({
+      'endPoint': 'https://<your-app-name>.skygeario.com/', // trailing slash is required
+      'apiKey': '<your-api-key>',
+    });
+    promise.then(()=> this.isConfigurated = true);
+    return promise;
+  }
+```
```

Comment:
Since we are suggesting user to configure the skygear in `ngOnInit`, which is guaranteed to run before application code. We do not need an promise interface `getSkygear`. Just a property getter will be easier to use?